src/esm_parser/yaml_to_dict.py

```diff
@@ -49,6 +55,106 @@ def __init__(self, fpath, yaml_error):
         super().__init__(self.message)
 
 
+def str_to_bool(string):
+    """Handy function to cast strings to bools. Eg.
+    bool('false') -> True
+    str_to_bool('false') -> False
+    str_to_bool('0') -> False
+    """
+    truthy_values = ["true", "1"]
+    return string.lower() in truthy_values
+
```

Comment:
This function already exist inside `esm_master`

Reply:
Do you mean parser? I could not find it in `esm_master`. Do you mean `could_be_bool` function in the parser? Unfortunately, I does not do what the function above can do. Eg.
```python
>>> could_be_bool(0)
False
```

or this one:
```python
>>> print(to_boolean(0))
None

>>> print(to_boolean('0'))
None
```

Reply:
By the way, a new solution is pushed so this one is obsolete now.

Reply:
https://github.com/esm-tools/esm_tools/blob/781c659c912c1bdefaecdff15e564b79c4a9e184/src/esm_parser/esm_parser.py#L2483-L2498

Also the `choose_` block have a way of evaluating 1 and 0 (any type) as booleans if the cases are True or False and the other way around.